salt/modules/mac_xattr.py

```diff
@@ -64,92 +70,116 @@ def read(path, attribute, hex=False):
     '''
     Read the given attributes on the given file/directory
 
-    CLI Example:
+    :param str path: The file to get attributes from
 
-    .. code-block:: bash
+    :param str attribute: The attribute to read
+
+    :param bool hex: return the values with forced hexadecimal values
 
-        salt '*' xattr.print /path/to/file com.test.attr
-        salt '*' xattr.list /path/to/file com.test.attr hex=True
+    CLI Example:
 
-    path
-        The file(s) to get the attribute from
+    .. code-block:: bash
 
-    hex
-        Should the value be returned with forced hexadecimal value?
+        salt '*' xattr.read /path/to/file com.test.attr
+        salt '*' xattr.read /path/to/file com.test.attr hex=True
     '''
     hex_flag = ""
     if hex:
         hex_flag = "-x"
+
     cmd = 'xattr -p {0} "{1}" "{2}"'.format(hex_flag, attribute, path)
-    return __salt__['cmd.run'](cmd)
+
+    try:
+        ret = salt.utils.mac_utils.execute_return_result(cmd)
+    except CommandExecutionError as exc:
+        if 'No such file' in exc.strerror:
+            raise CommandExecutionError('File not found: {0}'.format(path))
+        if 'No such xattr' in exc.strerror:
+            raise CommandExecutionError('Attribute not found: {0}'.format(attribute))
+        return 'Unknown Error: {0}'.format(exc.strerror)
```

Comment:
This also directly contradicts the type of return you have documented that this function will return.